### PR TITLE
Feature/NRediSearch-TAGS

### PR DESCRIPTION
**Pull request by rajeshsubhankar**
Redisearch **TAG** feature has been added. Now user can create schema with TAG fields and separators. This PR also includes **FT.TAGVALS** to get all distinct values in a TAG field.

[TAGS](http://redisearch.io./Tags/)
[tagvals](http://redisearch.io/Commands/#fttagvals)

### CODE REVIEW

File: NRediSearch/Client.cs [hunk 1]
```diff
@@ -138,6 +138,39 @@ public SearchResult Search(Query q)
...
+        /// </summary>
+        /// <param name="fieldName">TAG field name</param>
+        /// <returns>List of TAG field values</returns>
+        public RedisValue[] TagVals(string fieldName)
+        {
+            var args = new List<object>();
+            args.Add(_boxedIndexName);
+            args.Add(fieldName);
+
+            var resp = (RedisValue[])DbSync.Execute("FT.TAGVALS", args);
```
Comment (mgravell) [inline NRediSearch/Client.cs]: if the number of args is known and fixed: you don't need to use a list - there should be a `params object[]` overload, which means this should work?

    var resp = (RedisValue[])DbSync.Execute("FT.TAGVALS", _boxedIndexName, fieldName);

File: NRediSearch/Schema.cs [hunk 2]
```diff
@@ -15,7 +15,8 @@ public enum FieldType
         {
             FullText,
             Geo,
-            Numeric
+            Numeric,
+            TAG
```
Comment (mgravell) [inline NRediSearch/Schema.cs]: is tag an abbreviation? or is this literally a "tag"? If the latter: `Tag` (capitalization) would be preferred

File: NRediSearch/Schema.cs [hunk 3]
```diff
@@ -129,5 +131,36 @@ public Schema AddSortableNumericField(string name)
...
+        {
+            public string Separator { get; }
+            internal TagField(string name, string separator = ",") : base(name, FieldType.TAG, false)
+            {
+                Separator = separator;
+            }
+            internal override void SerializeRedisArgs(List<object> args)
+            {
+                base.SerializeRedisArgs(args);
+                if (!Separator.Equals(","))
```
Comment (mgravell) [inline NRediSearch/Schema.cs]: would prefer `if(Separator != ",")` here - avoids a possible NRE; note that C# has simpler operator usage than Java, so `==` does the obvious / correct thing

**Comment by mgravell** [comment]
looks good; I've added some feedback

**Comment by mgravell** [comment]
oof; `@@ -1,453 +1,445 @@` - looks like you did a ctrl+k,ctrl+d ? whitespace / document reformats are *fine*, but it is usually a good idea to keep those in a separate commit with *just* a reformat it in - it is virtually impossible to see the actual code changes in there now :(

**Comment by mgravell** [comment]
hmmm; on second glance I'm not quite sure *what* happened; git isn't very happy with it, but it might not have been you; I'll try to wade through the red...

**Comment by mgravell** [comment]
LGTM; merged; I can try to build/deploy later - thanks

**Comment by rajeshsubhankar** [comment]
Thanks a lot.

**Comment by dvirsky** [comment]
@mgravell can you please release a version of the client with this PR merged? Thanks.

**Comment by rajeshsubhankar** [comment]
@dvirsky We have one more pending PR @ https://github.com/StackExchange/StackExchange.Redis/pull/754

It will be great if we can include that as well. Thanks.

**Comment by mgravell** [comment]
Can you please confirm that 7 is the right flag? This is UseTermOffsets  | KeepFieldFlags  | UseScoreIndexes.

Are you sure this shouldn't be 8?

**Comment by mgravell** [comment]
@rajeshsubhankar  k; I've sought clarification on that #754, but yes: let's get that merged (when clarified) and deployed

**Pull request merged**